### PR TITLE
2007 Added Take 2

### PR DESCRIPTION
**Pull request by s3yoonpark**

### CODE REVIEW

File: solutions/bio1/StringRewriting/index.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { Link, SCodeBlock, SList, SText, STitle } from "components";
+import { brace, floor } from "solutions/util";
```
Comment (myst-6) [inline solutions/bio1/StringRewriting/index.tsx:2]: Are these unused imports?

Comment (s3yoonpark) [reply solutions/bio1/StringRewriting/index.tsx:2]: I just copied them from another person's. Is that what's causing the problem?

Comment (macaquedev) [reply solutions/bio1/StringRewriting/index.tsx:2]: yes. check which ones are not used and remove them please

Comment (s3yoonpark) [reply solutions/bio1/StringRewriting/index.tsx]: I removed all unused ones. It's still failing 😢

File: content/problems/bio1/index.tsx [hunk 2]
```diff
@@ -893,4 +894,14 @@ export const bio1Problems: BIO1ProblemInfo[] = [
     solutionAuthors: ["Adwaya Gupta"],
     component: <Trail />,
   },
+  {
+    difficulty: "Insane",
+    display: "String Rewriting",
+    original: "https://olympiad.org.uk/papers/2007/bio/bio07exam.pdf",
+    year: 2007,
+    tags: ["Q3", "Dynamic Programming", "Implementation", "Graphs"],
```
Comment (myst-6) [inline content/problems/bio1/index.tsx:902]: Q3 doesn't go as a tag anymore. `question` is an attribute on the object itself.

Comment (s3yoonpark) [reply content/problems/bio1/index.tsx]: I fixed it. Why is it still jambloating 😭 AAAAA

File: solutions/bio1/StringRewriting/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        This means a divide-and-conquer approach could be used for this question. 
+      </SText>
+
+      <STitle>Solution</STitle>
+      <SText>
+        Initially, we precompute 2 sequences and store them into arrays: powers of 2 and the fibonacci sequence. This will allow $O(1)$ retrieval for further queries regarding these 2 sequences. 
+      </SText>
+      <SText>
+        We will store the answer in a vector with size 5 called $ans$, initially filled with 0s. 
+        We will also store the length has been processed so far in $len$. By keeping track of this variable, we can ensure the information contributing to $ans$ is only when $len <= pos$.
```
Comment (myst-6) [inline solutions/bio1/StringRewriting/index.tsx:31]: Need to use \le instead of <=

Comment (s3yoonpark) [reply solutions/bio1/StringRewriting/index.tsx]: Thanks so much!! 
@myst-6 @macaquedev The preview is deployed now.

**Review by macaquedev — APPROVED** [review]
yayyy

**Pull request merged**